src/components/dialogs/network-modifications/tabular-modification/tabular-modification-form.tsx

```diff
@@ -0,0 +1,199 @@
+                    postProcessFile(results.data);
+                },
+            });
+        }
+    }, [postProcessFile, selectedFile, selectedFileError, setValue]);
+
+    const typesOptions = useMemo(() => {
+        const equipmentTypesToExclude = new Set([
+            EQUIPMENT_TYPES.SWITCH,
+            EQUIPMENT_TYPES.LCC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.VSC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.HVDC_CONVERTER_STATION,
+            EQUIPMENT_TYPES.STATIC_VAR_COMPENSATOR,
+            EQUIPMENT_TYPES.DANGLING_LINE,
+            EQUIPMENT_TYPES.THREE_WINDINGS_TRANSFORMER,
+            EQUIPMENT_TYPES.HVDC_LINE,
+            EQUIPMENT_TYPES.SHUNT_COMPENSATOR,
+            EQUIPMENT_TYPES.SUBSTATION,
+            EQUIPMENT_TYPES.VOLTAGE_LEVEL,
+            EQUIPMENT_TYPES.TWO_WINDINGS_TRANSFORMER,
+            EQUIPMENT_TYPES.LINE,
+            EQUIPMENT_TYPES.BATTERY,
+        ]);
+        return Object.values(EQUIPMENT_TYPES).filter(
+            (equipmentType) => !equipmentTypesToExclude.has(equipmentType)
+        );
+    }, []);
+
+    const handleChange = useCallback(() => {
+        setTypeChanged(!typeChanged);
+        setValue(MODIFICATIONS_TABLE, []);
+    }, [setValue, typeChanged]);
+
+    const equipmentTypeField = (
+        <AutocompleteInput
+            isOptionEqualToValue={richTypeEquals}
+            name={TYPE}
+            label="Type"
+            options={typesOptions}
+            onChangeCallback={handleChange}
+            getOptionLabel={richTypeLabel}
+            size={'small'}
+            formProps={{ variant: 'filled' }}
+        />
+    );
+
+    const defaultColDef = useMemo(
+        () => ({
+            flex: 1,
+            filter: true,
+            sortable: true,
+            resizable: false,
+            lockPinned: true,
+            wrapHeaderText: true,
+            autoHeaderHeight: true,
+            cellRenderer: DefaultCellRenderer,
+        }),
+        []
+    );
+    interface ColumnDefinition {
+        [key: string]: any;
+    }
+
+    const columnDefs = useMemo(() => {
+        return TABULAR_MODIFICATION_FIELDS[watchType]?.map((field) => {
+            let colunmDef: ColumnDefinition = {};
+            if (field === 'equipmentId') {
+                colunmDef.pinned = true;
+            }
+            colunmDef.field = intl.formatMessage({ id: field });
+            colunmDef.headerName = intl.formatMessage({ id: field });
+            colunmDef.cellRenderer = DefaultCellRenderer;
+            return colunmDef;
+        });
+    }, [intl, watchType]);
+
+    return (
+        <>
+            <Grid container spacing={2}>
+                <Grid container item xs={12} spacing={2}>
+                    {gridItem(equipmentTypeField, 4)}
+                    <Grid item sx={styles.csvButton}>
+                        {FileField}
+                    </Grid>
+                </Grid>
+                <Grid item sx={styles.csvButton}>
```

Comment:
is styles.csvButton necessary as you are using spacing={2}  in the parent to add the gap?